concourse/steps/update_component_deps.py

```diff
@@ -503,19 +691,47 @@ def create_upgrade_pr(
             '\n\nRelease notes were shortened since they exceeded the maximum length allowed for a '
             'pull request body. The remaining release notes will be added as comments to this PR.'
         )
-        if max_pr_body_length < len(release_notes):
-            step_size = max_pr_body_length - len(max_body_length_exceeded_remark)
+
+        # Split the formatted_diff if it exceeds max body length (line-by-line)
+        if len(formatted_diff) > max_pr_body_length:
```

Comment:
why don't you simply calculate the allowed length, and shorten the diff accordingly, like so:

```
max_leng = allowed - len(diff)
diff = diff[:len(diff + max_leng) - 3] + '..'
```

Assuming we wanted to remove entire lines (why?), it would be preferable to calculate location of last line-break (by looking for linebreak backwards from last allowed character). That way, we less str-operations, and also do not need to duplicate the strs.

However, even better than just cutting diff might be generating a shorter version (e.g. only print components and omit resource-changes).

There should also rather be a better indication of shortening than the `..` I used above as a short example, to make it obvious to readers that a portion of the diff is missing.